zjb/templates/results.html.j2

```diff
@@ -36,4 +39,86 @@
+                    url = '/fetch-from-database' + params
+                    textbox.value = getValueFromRequest(url, project, branch, pipeline, job)
+
+
+
+
+                    const saveButton = document.createElement('button');
+                    saveButton.textContent = 'Save';
+                    saveButton.onclick = function() {
+                        url = '/save-to-database' + params
+                        saveToDatabase(url, project, branch, pipeline, job, textbox.value);
+                        textbox.parentNode.removeChild(textbox);
+                        saveButton.parentNode.removeChild(saveButton);
+                    };
+                    
+                    const cancelButton = document.createElement('button');
+                    cancelButton.textContent = 'Cancel';
+                    cancelButton.onclick = function() {
+                        textbox.parentNode.removeChild(textbox);
+                        saveButton.parentNode.removeChild(saveButton);
+                    };
+                }
+
+                function saveToDatabase(url, project, branch, pipeline, job, note) {
+                    var xhr = new XMLHttpRequest();
+                    xhr.open('POST', url);
+                    xhr.setRequestHeader('Content-Type', 'application/json');
+                    xhr.onload = function() {
+                        if (xhr.status === 200) {
+                            console.log('Data saved to database!');
+                        } else {
+                            console.log('Error saving data to database.');
+                        }
+                    };
+
+                    let data = {
+                        url: url,
+                        project: project,
+                        branch: branch,
+                        pipeline: pipeline,
+                        job: job,
+                        note: note
+                    };
+                    xhr.send(JSON.stringify(data));
+                }
+
+                function getValueFromRequest(url, project, branch, pipeline, job) {
+                    return new Promise((resolve, reject) => {
+                        const xhr = new XMLHttpRequest();
+                        xhr.open('GET', url);
+                        xhr.onload = () => {
+                        if (xhr.status === 200) {
+                            resolve(xhr.responseText);
+                        } else {
+                            reject(xhr.statusText);
+                        }
+                        };
+                        xhr.onerror = () => reject(xhr.statusText);
+
+                        let data = {
+                            url: url,
+                            project: project,
+                            branch: branch,
+                            pipeline: pipeline,
+                            job: job,
+                            note: note
```

Comment:
I get the following here: `caught (in promise) ReferenceError: note is not defined`